misc/config_tools/xforms/lib.xsl

```diff
@@ -0,0 +1,495 @@
+  <xsl:variable name="end_of_array_initializer" select="concat('};', $newline)" />
+  <xsl:variable name="else" select="concat('#else', $newline)" />
+  <xsl:variable name="endif" select="concat('#endif', $newline)" />
+  <xsl:variable name="license">
+    <xsl:text>/*&#xa;</xsl:text>
+    <xsl:text> * Copyright (C) 2021 Intel Corporation. All rights reserved.&#xa;</xsl:text>
+    <xsl:text> *&#xa;</xsl:text>
+    <xsl:text> * SPDX-License-Identifier: BSD-3-Clause&#xa;</xsl:text>
+    <xsl:text> */&#xa;</xsl:text>
+    <xsl:text>&#xa;</xsl:text>
+  </xsl:variable>
+  <!-- End of C code variables -->
+
+  <!-- Generic library auxiliaries-->
+  <!-- < Do not call auxiliaries directly externally and internally >-->
+  <!-- < Please call auxiliaries wrappers instead >-->
+  <func:function name="acrn:string-to-num-aux">
+    <xsl:param name="v" />
+    <xsl:param name="base" />
+    <xsl:param name="acc" />
+    <xsl:variable name="digit" select="string-length(substring-before($digits, substring($v, 1, 1)))" />
+    <xsl:choose>
+      <xsl:when test="string-length($v) = 1">
+        <func:result select="($acc * $base) + $digit" />
+      </xsl:when>
+      <xsl:otherwise>
+        <func:result select="acrn:string-to-num-aux(substring($v, 2), $base, ($acc * $base) + $digit)" />
+      </xsl:otherwise>
+    </xsl:choose>
+  </func:function>
+
+  <func:function name="acrn:find-list-min-aux">
+    <xsl:param name="list" />
+    <xsl:param name="delimar" />
+    <xsl:param name="min" />
+    <func:result>
+      <xsl:choose>
+        <xsl:when test="contains($list, $delimar)">
+          <xsl:value-of select="acrn:find-list-min-aux(substring-after($list, $delimar), $delimar, acrn:min(substring-before($list, $delimar), $min))" />
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:value-of select="acrn:min($list, $min)" />
+        </xsl:otherwise>
+      </xsl:choose>
+    </func:result>
+  </func:function>
+
+  <func:function name="acrn:convert-num-base-aux">
+    <xsl:param name="v" />
+    <xsl:param name="base" />
+    <xsl:param name="acc" />
+    <xsl:variable name="digit" select="acrn:num-to-string($v, $base)" />
+    <xsl:choose>
+      <xsl:when test="floor($v div $base) = 0">
+        <func:result select="concat($digit, $acc)" />
+      </xsl:when>
+      <xsl:otherwise>
+        <func:result select="acrn:convert-num-base-aux(floor($v div $base), $base, concat($digit, $acc))" />
+      </xsl:otherwise>
+    </xsl:choose>
+  </func:function>
+  <!-- End of generic library auxiliaries-->
+
+  <!-- Generic library routines-->
+  <func:function name="acrn:string-to-num">
+    <xsl:param name="v" />
+    <xsl:param name="base" />
+    <func:result select="acrn:string-to-num-aux($v, $base, 0)" />
+  </func:function>
+
+  <func:function name="acrn:num-to-string">
+    <xsl:param name="v" />
+    <xsl:param name="base" />
+    <func:result select="substring($digits, ($v mod $base) + 1 , 1)" />
```

Comment:
It seems this function only converts a number to its last digit. Is this what this function is intended to do?

Reply:
Fixed:
<acrn:convert-num-base> now is calling "acrn:string-to-num" and "acrn:num-to-string".
<acrn:num-to-string> now is calling "acrn:num-to-string-aux"
The acrn:convert-num-base-aux is renamed as "acrn:num-to-string-aux"

<func:function name="acrn:num-to-string"> is no longer converts last digit, it convert the whole string from decimal to new base number string.

Reply:
Fine with me.